interfaces/src/dotty/tools/dotc/interfaces/SimpleReporter.scala

```diff
@@ -0,0 +1,19 @@
+package dotty.tools.dotc.interfaces
+
+import scala.annotation.binaryCompatible
+
+
+/** Report errors, warnings and info messages during the compilation process
+ *
+ * You should implement this interface if you want to handle the diagnostics
+ * returned by the compiler yourself.
+ *
+ * See the method `process` of `dotty.tools.dotc.Driver` for more information.
+ */
+@binaryCompatible
+trait SimpleReporter {
```

Comment:
> default public void $init$() {
    }

Empty trait constructor methods are indeed checked by hand.

>   default public static /* synthetic */ void report$(SimpleReporter $this, Diagnostic diag) {
        $this.report(diag);
    }

why would it be emmited if there's no default implementation provided(this method is abstract)?